shared/src/business/useCases/practitioners/updatePractitionerUserInteractor.js

```diff
@@ -66,16 +96,51 @@ exports.updatePractitionerUserInteractor = async ({
     userId: requestUser.userId,
   });
 
-  await generateChangeOfAddress({
+  if (userHasAccount && userIsUpdatingEmail) {
+    await applicationContext.getUseCaseHelpers().sendEmailVerificationLink({
+      applicationContext,
+      pendingEmail: user.pendingEmail,
+      pendingEmailVerificationToken: user.pendingEmailVerificationToken,
+    });
+  }
+
+  const updatedPractitionerRaw = new Practitioner(updatedUser, {
     applicationContext,
-    bypassDocketEntry,
-    contactInfo: validatedUserData.contact,
-    requestUserId: requestUser.userId,
-    updatedEmail: validatedUserData.email,
-    updatedName: validatedUserData.name,
-    user: oldUserInfo,
-    websocketMessagePrefix: 'admin',
-  });
+  }).toRawObject();
+  const oldPractitionerRaw = new Practitioner(oldUserInfo, {
+    applicationContext,
+  }).toRawObject();
+
+  const practitionerDetailDiff = applicationContext
+    .getUtilities()
+    .getAddressPhoneDiff({
+      newData: {
+        ...omit(updatedPractitionerRaw, 'contact'),
+        ...updatedPractitionerRaw.contact,
+      },
+      oldData: {
+        ...omit(oldPractitionerRaw, 'contact'),
+        ...oldPractitionerRaw.contact,
+      },
+    });
+
+  const hasUpdatedEmailOnly = isEqual(
+    Object.keys(practitionerDetailDiff).sort(),
+    ['pendingEmail', 'pendingEmailVerificationToken'],
+  );
+
+  if (!hasUpdatedEmailOnly) {
```

Comment:
I dig this, but perhaps it could be put into a reusable function and not clutter up this one to only figure out a boolean.

Reply:
I've taken the general comparison code into its own function.